environment.go

```diff
@@ -57,9 +120,65 @@ func (e *PipelineEnvironment) exportEnvironment(path string) error {
 
 func (e *PipelineEnvironment) createTemplateParser() *template.Template {
 	fm := template.FuncMap{}
+	// {{ Key }}
+	// Required environment value, if not defined it will not be found as function and raise an error
 	for k, v := range e.Environment {
-		fm[k] = func() string { return *v }
+		// Convert string pointer to static string
+		fm[k] = func(v string) func() (string, error) {
+			return func() (string, error) {
+				return v, nil
+			}
+		}(*v)
+	}
+	// {{ Env "Key" ["Default"] }}
+	// Usable as optional environment variable, can provide default value if not defined
+	fm["Env"] = func(args ...interface{}) (string, error) {
+		if len(args) < 1 {
+			return "", errors.New(fmt.Sprintf("Env: missing argument(s). Need atleast 1 argument"))
+		}
+		if len(args) > 2 {
+			return "", errors.New(fmt.Sprintf("Env: too many arguments. Got %d want <=2", len(args)))
+		}
+		parts := make([]string, len(args))
+		for i, v := range args {
+			parts[i] = fmt.Sprint(v)
+		}
+		val, ok := e.Environment[parts[0]]
+		if !ok {
+			if len(parts) < 2 {
+				return "", errors.New(fmt.Sprintf("Env '%s' not defined, no fallback provided", parts[0]))
+			}
+			return parts[1], nil
+		}
+		return *val, nil
 	}
+	// {{ EnvDir "Key" ["Default"] }}
+	// Get Path from environment, converts to absolute path using filepath.Abs
```

Comment:
I think this is highly non obvious. In most template engines I know this would be `{{ EnvDir "Key" "Default" | AbsPath}}` or similar

Reply:
Given an `AbsPath` function is defined `{{ EnvDir "Key" "Default" }}` would be equivalent to `{{ Env "Key" "Default" | AbsPath}}`.